master/internal/api_model.go

```diff
@@ -298,6 +288,10 @@ func (a *apiServer) PostModel(
 		reqLabels, req.Notes, user.User.Id, workspaceID,
 	)
 
+	if err != nil && strings.Contains(err.Error(), "violates unique constraint") {
```

Comment:
maybe we could probably use a shared way of detecting these errors. let's see what we do with the other pr wrt using db err codes. 
we have some utils in the `api` package that translate internal errors to api facing ones maybe this would be a good new case there